networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py

```diff
@@ -254,11 +254,18 @@ def allocate_hosting_port(self, context, router_id, port_db, network_type,
         # If this is a router interface, the VLAN comes from APIC.
         # If it's the gateway, the VLAN comes from the segment ID
         if port_db.get('device_owner') == DEVICE_OWNER_ROUTER_GW:
-            return super(AciVLANTrunkingPlugDriver,
-                         self).allocate_hosting_port(
-                             context, router_id,
-                             port_db, network_type, hosting_device_id
-                         )
+            ext_dict, net = self._get_external_network_dict(context, port_db)
+            # If an OpFlex network is used on the external network,
+            # the actual segment ID comes from the confgi file
+            if net.get('provider:network_type') == 'opflex':
+                if ext_dict.get('segmentation_id'):
+                    return {'allocated_port_id': port_db.id,
+                            'allocated_vlan': ext_dict['segmentation_id']}
```

Comment:
as we discussed, if 'segmentation_id' is not present, then a proper exception, error log should be given.